### PR TITLE
[DataGrid] Improve column header separators

### LINKED ISSUE #7268: [data grid] Improve column separators to display ability to resize columns
### Summary 💡

The column headers separators present no distinguishable difference between columns that can be resized and columns that don't. This is a follow-up on #1623.

On https://github.com/mui/mui-x/pull/7067, we started exploring ways to make this difference clear, but there're also some [challenges to validate](https://github.com/mui/mui-x/pull/7067#issuecomment-1356745665) and other solutions to explore.

### Examples 🌈

image

### Motivation 🔦

- Discoverability and identification of resizable columns (Better UX).

### Design
- [Figma Process](https://www.figma.com/file/W6TItYkiGJZjL5zLnfeSjy/Data-grid?node-id=1227%3A62449&t=PrKs3y9Cx0tIZdmK-0)
- [Prototype](https://www.figma.com/proto/W6TItYkiGJZjL5zLnfeSjy/Data-grid?node-id=1367%3A64345&scaling=contain&page-id=1227%3A62449&starting-point-node-id=1367%3A64345)

### PR DESCRIPTION
**Pull request by MBilalShafi**
Fixes #7268 

Comparison of different aspects: https://codesandbox.io/s/purple-shadow-c8l2xk?file=/demo.tsx

### CODE REVIEW

File: docs/pages/x/api/data-grid/data-grid-premium.json [hunk 1]
```diff
@@ -445,7 +445,6 @@
       "columnHeaders",
       "columnHeadersInner",
       "columnHeadersInner--scrollable",
-      "columnSeparator--resizable",
```
Comment (MBilalShafi) [inline docs/pages/x/api/data-grid/data-grid-premium.json:448]: Or possibly keep this class until v7 for being non-breaking?

**Comment by cherniavskii** [comment]
I have merged a slightly related PR recently - https://github.com/mui/mui-x/pull/9076 (it's also the one that caused conflicts for this PR, sorry about that 😅)
Should we show the column header borders on mobile by default (without hover)?

**Comment by MBilalShafi** [comment]
>  Should we show the column header borders on mobile by default (without hover)?

I think it makes sense since we enabled other hover actions on mobile by default.

**Comment by cherniavskii** [comment]
> Not sure about the resize handle in the hovered state though if it'll look too heavier 🤔

We don't have to enable all hover styles by default on mobile, but only those applied on the column header cell level, so that we avoid issues like https://github.com/mui/mui-x/issues/8762 and https://github.com/mui/mui-x/issues/9073

**Comment by MBilalShafi** [comment]
Handled in https://github.com/mui/mui-x/pull/14293

**Pull request closed without merging**